aten/src/ATen/native/mps/operations/Indexing.mm

```diff
@@ -710,10 +710,23 @@ Tensor index_select_mps(const Tensor & self,
 
   MPSGraphCache* cache_ = MPSGraphCache::getInstance();
 
+  MPSDataType inputDataType = getMPSScalarType(self.scalar_type());
+  MPSDataType maskDataType = getMPSScalarType(b_mask->scalar_type());
+  // Workaround for `selectWithPredicateTensor` on macOS Monterey where bool data type may cause a hang
+  // The issue is fixed in macOS Ventura (13.0)
```

Comment:
Why not get rid of the OS check and have Int8 for older OS too ?

Reply:
This check is only for older OSes (PyTorch MPS backend was added starting with macOS Monterey). On Ventura this is already fixed.